From 7ab33bf335e3e5b5ae9e3d507c78d108d40bb94a Mon Sep 17 00:00:00 2001
From: David Trasbo <thisisdato@gmail.com>
Date: Wed, 19 Aug 2009 19:27:08 +0200
Subject: [PATCH] Reorganized application tests.

  test/unit => test/models
  test/unit/helpers => test/helpers
  test/unit/**/*_observer_test.rb => test/observers
  test/functional + test/integration => test/controllers

Generators and Rake tasks has been updated accordingly. Additionally an update:test_directory task was added for backwards compatibility.
---
 .../test_unit/controller/controller_generator.rb   |    2 +-
 .../test_unit/helper/helper_generator.rb           |    2 +-
 .../test_unit/integration/integration_generator.rb |    2 +-
 .../test_unit/mailer/mailer_generator.rb           |    2 +-
 .../generators/test_unit/model/model_generator.rb  |    2 +-
 .../test_unit/observer/observer_generator.rb       |    2 +-
 .../test_unit/scaffold/scaffold_generator.rb       |    2 +-
 railties/lib/tasks/framework.rake                  |   30 ++++++++++++++++-
 railties/lib/tasks/testing.rake                    |   36 +++++++++++--------
 railties/test/generators/app_generator_test.rb     |    6 ++--
 .../test/generators/controller_generator_test.rb   |    8 ++--
 railties/test/generators/helper_generator_test.rb  |    2 +-
 .../generators/integration_test_generator_test.rb  |    2 +-
 railties/test/generators/mailer_generator_test.rb  |    2 +-
 railties/test/generators/model_generator_test.rb   |    2 +-
 .../test/generators/observer_generator_test.rb     |    2 +-
 .../test/generators/resource_generator_test.rb     |   10 +++---
 .../scaffold_controller_generator_test.rb          |    8 ++--
 .../test/generators/scaffold_generator_test.rb     |   12 +++---
 19 files changed, 84 insertions(+), 50 deletions(-)
 create mode 100644 railties/lib/generators/rails/app/templates/test/controllers/.empty_directory
 delete mode 100644 railties/lib/generators/rails/app/templates/test/functional/.empty_directory
 create mode 100644 railties/lib/generators/rails/app/templates/test/helpers/.empty_directory
 delete mode 100644 railties/lib/generators/rails/app/templates/test/integration/.empty_directory
 create mode 100644 railties/lib/generators/rails/app/templates/test/models/.empty_directory
 delete mode 100644 railties/lib/generators/rails/app/templates/test/unit/.empty_directory

diff --git a/railties/lib/generators/rails/app/templates/test/controllers/.empty_directory b/railties/lib/generators/rails/app/templates/test/controllers/.empty_directory
new file mode 100644
index 0000000..e69de29
diff --git a/railties/lib/generators/rails/app/templates/test/functional/.empty_directory b/railties/lib/generators/rails/app/templates/test/functional/.empty_directory
deleted file mode 100644
index e69de29..0000000
diff --git a/railties/lib/generators/rails/app/templates/test/helpers/.empty_directory b/railties/lib/generators/rails/app/templates/test/helpers/.empty_directory
new file mode 100644
index 0000000..e69de29
diff --git a/railties/lib/generators/rails/app/templates/test/integration/.empty_directory b/railties/lib/generators/rails/app/templates/test/integration/.empty_directory
deleted file mode 100644
index e69de29..0000000
diff --git a/railties/lib/generators/rails/app/templates/test/models/.empty_directory b/railties/lib/generators/rails/app/templates/test/models/.empty_directory
new file mode 100644
index 0000000..e69de29
diff --git a/railties/lib/generators/rails/app/templates/test/unit/.empty_directory b/railties/lib/generators/rails/app/templates/test/unit/.empty_directory
deleted file mode 100644
index e69de29..0000000
diff --git a/railties/lib/generators/test_unit/controller/controller_generator.rb b/railties/lib/generators/test_unit/controller/controller_generator.rb
index b57a6e7..27ea9af 100644
--- a/railties/lib/generators/test_unit/controller/controller_generator.rb
+++ b/railties/lib/generators/test_unit/controller/controller_generator.rb
@@ -7,7 +7,7 @@ module TestUnit
 
       def create_test_files
         template 'functional_test.rb',
-                 File.join('test/functional', class_path, "#{file_name}_controller_test.rb")
+                 File.join('test/controllers', class_path, "#{file_name}_controller_test.rb")
       end
     end
   end
diff --git a/railties/lib/generators/test_unit/helper/helper_generator.rb b/railties/lib/generators/test_unit/helper/helper_generator.rb
index 9ecfaa4..e07dfe0 100644
--- a/railties/lib/generators/test_unit/helper/helper_generator.rb
+++ b/railties/lib/generators/test_unit/helper/helper_generator.rb
@@ -6,7 +6,7 @@ module TestUnit
       check_class_collision :suffix => "HelperTest"
 
       def create_helper_files
-        template 'helper_test.rb', File.join('test/unit/helpers', class_path, "#{file_name}_helper_test.rb")
+        template 'helper_test.rb', File.join('test/helpers', class_path, "#{file_name}_helper_test.rb")
       end
     end
   end
diff --git a/railties/lib/generators/test_unit/integration/integration_generator.rb b/railties/lib/generators/test_unit/integration/integration_generator.rb
index d9d9b3b..5be4267 100644
--- a/railties/lib/generators/test_unit/integration/integration_generator.rb
+++ b/railties/lib/generators/test_unit/integration/integration_generator.rb
@@ -6,7 +6,7 @@ module TestUnit
       check_class_collision :suffix => "Test"
 
       def create_test_files
-        template 'integration_test.rb', File.join('test/integration', class_path, "#{file_name}_test.rb")
+        template 'integration_test.rb', File.join('test/controllers', class_path, "#{file_name}_test.rb")
       end
     end
   end
diff --git a/railties/lib/generators/test_unit/mailer/mailer_generator.rb b/railties/lib/generators/test_unit/mailer/mailer_generator.rb
index ef350a6..f138f71 100644
--- a/railties/lib/generators/test_unit/mailer/mailer_generator.rb
+++ b/railties/lib/generators/test_unit/mailer/mailer_generator.rb
@@ -7,7 +7,7 @@ module TestUnit
       check_class_collision :suffix => "Test"
 
       def create_test_files
-        template "unit_test.rb", File.join('test/unit', class_path, "#{file_name}_test.rb")
+        template "unit_test.rb", File.join('test/models', class_path, "#{file_name}_test.rb")
       end
 
       def create_fixtures_files
diff --git a/railties/lib/generators/test_unit/model/model_generator.rb b/railties/lib/generators/test_unit/model/model_generator.rb
index 469306e..fe5c521 100644
--- a/railties/lib/generators/test_unit/model/model_generator.rb
+++ b/railties/lib/generators/test_unit/model/model_generator.rb
@@ -9,7 +9,7 @@ module TestUnit
       check_class_collision :suffix => "Test"
 
       def create_test_file
-        template 'unit_test.rb', File.join('test/unit', class_path, "#{file_name}_test.rb")
+        template 'unit_test.rb', File.join('test/models', class_path, "#{file_name}_test.rb")
       end
 
       hook_for :fixture_replacement
diff --git a/railties/lib/generators/test_unit/observer/observer_generator.rb b/railties/lib/generators/test_unit/observer/observer_generator.rb
index 14181f4..8d24177 100644
--- a/railties/lib/generators/test_unit/observer/observer_generator.rb
+++ b/railties/lib/generators/test_unit/observer/observer_generator.rb
@@ -6,7 +6,7 @@ module TestUnit
       check_class_collision :suffix => "ObserverTest"
 
       def create_test_files
-        template 'unit_test.rb',  File.join('test/unit', class_path, "#{file_name}_observer_test.rb")
+        template 'unit_test.rb',  File.join('test/observers', class_path, "#{file_name}_observer_test.rb")
       end
     end
   end
diff --git a/railties/lib/generators/test_unit/scaffold/scaffold_generator.rb b/railties/lib/generators/test_unit/scaffold/scaffold_generator.rb
index 78fcea1..262bed9 100644
--- a/railties/lib/generators/test_unit/scaffold/scaffold_generator.rb
+++ b/railties/lib/generators/test_unit/scaffold/scaffold_generator.rb
@@ -10,7 +10,7 @@ module TestUnit
 
       def create_test_files
         template 'functional_test.rb',
-                 File.join('test/functional', controller_class_path, "#{controller_file_name}_controller_test.rb")
+                 File.join('test/controllers', controller_class_path, "#{controller_file_name}_controller_test.rb")
       end
     end
   end
diff --git a/railties/lib/tasks/framework.rake b/railties/lib/tasks/framework.rake
index a084af7..fad94c1 100644
--- a/railties/lib/tasks/framework.rake
+++ b/railties/lib/tasks/framework.rake
@@ -78,7 +78,7 @@ namespace :rails do
   end
 
   desc "Update both configs, scripts and public/javascripts from Rails"
-  task :update => [ "update:configs", "update:javascripts", "update:scripts", "update:application_controller" ]
+  task :update => [ "update:configs", "update:javascripts", "update:scripts", "update:application_controller", "update:test_directory" ]
 
   desc "Applies the template supplied by LOCATION=/path/to/template"
   task :template do
@@ -129,5 +129,33 @@ namespace :rails do
         puts "#{old_style} has been renamed to #{new_style}, update your SCM as necessary"
       end
     end
+
+    desc "Move test directories to new locations"
+    task :test_directory do
+      if File.exists?(RAILS_ROOT + '/test')
+        FileUtils.mkdir(RAILS_ROOT + '/test/controllers') unless File.exists?(RAILS_ROOT + '/test/controllers')
+        [RAILS_ROOT + '/test/functional', RAILS_ROOT + '/test/integration'].each do |controller_test_dir|
+          if File.exists?(controller_test_dir)
+            FileUtils.mv(Dir["#{controller_test_dir}/**/*"], RAILS_ROOT + '/test/controllers', :force => true)
+            FileUtils.rm_rf(controller_test_dir)
+          end
+        end
+
+        if File.exists?(RAILS_ROOT + '/test/unit/helpers')
+          FileUtils.mv(RAILS_ROOT + '/test/unit/helpers', RAILS_ROOT + '/test/helpers')
+        else
+          FileUtils.mkdir(RAILS_ROOT + '/test/helpers') unless File.exists?(RAILS_ROOT + '/test/helpers')
+        end
+
+        if File.exists?(RAILS_ROOT + '/test/unit')
+          observer_tests = Dir["#{RAILS_ROOT}/test/unit/**/*_observer_test.rb"]
+          unless observer_tests.empty?
+            FileUtils.mkdir(RAILS_ROOT + '/test/observers')
+            FileUtils.mv(observer_tests, RAILS_ROOT + '/test/observers', :force => true)
+          end
+          FileUtils.mv(RAILS_ROOT + '/test/unit', RAILS_ROOT + '/test/models')
+        end
+      end
+    end
   end
 end
diff --git a/railties/lib/tasks/testing.rake b/railties/lib/tasks/testing.rake
index fd5e52a..9f48665 100644
--- a/railties/lib/tasks/testing.rake
+++ b/railties/lib/tasks/testing.rake
@@ -40,7 +40,7 @@ end
 
 desc 'Run all unit, functional and integration tests'
 task :test do
-  errors = %w(test:units test:functionals test:integration).collect do |task|
+  errors = %w(test:models test:controllers test:helpers).collect do |task|
     begin
       Rake::Task[task].invoke
       nil
@@ -55,8 +55,10 @@ namespace :test do
   Rake::TestTask.new(:recent => "db:test:prepare") do |t|
     since = TEST_CHANGES_SINCE
     touched = FileList['test/**/*_test.rb'].select { |path| File.mtime(path) > since } +
-      recent_tests('app/models/**/*.rb', 'test/unit', since) +
-      recent_tests('app/controllers/**/*.rb', 'test/functional', since)
+      recent_tests('app/models/**/*.rb', 'test/models', since) +
+      recent_tests('app/models/**/*_observer.rb', 'test/observers', since) +
+      recent_tests('app/controllers/**/*.rb', 'test/controllers', since) +
+      recent_tests('app/helpers/**/*.rb', 'test/helpers', since)
 
     t.libs << 'test'
     t.verbose = true
@@ -75,12 +77,16 @@ namespace :test do
       end
 
       models      = changed_since_checkin.select { |path| path =~ /app[\\\/]models[\\\/].*\.rb$/ }
+      observers   = changed_since_checkin.select { |path| path =~ /app[\\\/]models[\\\/].*_observer\.rb$/ }
       controllers = changed_since_checkin.select { |path| path =~ /app[\\\/]controllers[\\\/].*\.rb$/ }
+      helpers     = changed_since_checkin.select { |path| path =~ /app[\\\/]helpers[\\\/].*\.rb$/ }
 
-      unit_tests       = models.map { |model| "test/unit/#{File.basename(model, '.rb')}_test.rb" }
-      functional_tests = controllers.map { |controller| "test/functional/#{File.basename(controller, '.rb')}_test.rb" }
+      model_tests      = models.map { |model| "test/models/#{File.basename(model, '.rb')}_test.rb" }
+      observer_tests   = observers.map { |observer| "test/observers/#{File.basename(observer, '.rb')}_test.rb" }
+      controller_tests = controllers.map { |controller| "test/controllers/#{File.basename(controller, '.rb')}_test.rb" }
+      helper_tests     = helpers.map { |helper| "test/helpers/#{File.basename(helper, '.rb')}_test.rb" }
 
-      unit_tests.uniq + functional_tests.uniq
+      (model_tests + observer_tests + controller_tests + helper_tests).uniq
     end
 
     t.libs << 'test'
@@ -88,26 +94,26 @@ namespace :test do
   end
   Rake::Task['test:uncommitted'].comment = "Test changes since last checkin (only Subversion and Git)"
 
-  Rake::TestTask.new(:units => "db:test:prepare") do |t|
+  Rake::TestTask.new(:models => "db:test:prepare") do |t|
     t.libs << "test"
-    t.pattern = 'test/unit/**/*_test.rb'
+    t.pattern = ['test/models/**/*_test.rb', 'test/observers/**/*_test.rb']
     t.verbose = true
   end
-  Rake::Task['test:units'].comment = "Run the unit tests in test/unit"
+  Rake::Task['test:models'].comment = "Run the unit tests in test/models and test/observers"
 
-  Rake::TestTask.new(:functionals => "db:test:prepare") do |t|
+  Rake::TestTask.new(:controllers => "db:test:prepare") do |t|
     t.libs << "test"
-    t.pattern = 'test/functional/**/*_test.rb'
+    t.pattern = 'test/controllers/**/*_test.rb'
     t.verbose = true
   end
-  Rake::Task['test:functionals'].comment = "Run the functional tests in test/functional"
+  Rake::Task['test:controllers'].comment = "Run the functional and integration tests in test/controllers"
 
-  Rake::TestTask.new(:integration => "db:test:prepare") do |t|
+  Rake::TestTask.new(:helpers => "db:test:prepare") do |t|
     t.libs << "test"
-    t.pattern = 'test/integration/**/*_test.rb'
+    t.pattern = 'test/helpers/**/*_test.rb'
     t.verbose = true
   end
-  Rake::Task['test:integration'].comment = "Run the integration tests in test/integration"
+  Rake::Task['test:helpers'].comment = "Run the unit tests in test/helpers"
 
   Rake::TestTask.new(:benchmark => 'db:test:prepare') do |t|
     t.libs << 'test'
diff --git a/railties/test/generators/app_generator_test.rb b/railties/test/generators/app_generator_test.rb
index 19e41c1..566c0dd 100644
--- a/railties/test/generators/app_generator_test.rb
+++ b/railties/test/generators/app_generator_test.rb
@@ -34,11 +34,11 @@ class AppGeneratorTest < GeneratorsTestCase
       public/javascripts
       public/stylesheets
       script/performance
+      test/controllers
       test/fixtures
-      test/functional
-      test/integration
+      test/helpers
+      test/models
       test/performance
-      test/unit
       vendor
       vendor/plugins
       tmp/sessions
diff --git a/railties/test/generators/controller_generator_test.rb b/railties/test/generators/controller_generator_test.rb
index 850b9e6..bb3e49c 100644
--- a/railties/test/generators/controller_generator_test.rb
+++ b/railties/test/generators/controller_generator_test.rb
@@ -27,23 +27,23 @@ class ControllerGeneratorTest < GeneratorsTestCase
   def test_invokes_helper
     run_generator
     assert_file "app/helpers/account_helper.rb"
-    assert_file "test/unit/helpers/account_helper_test.rb"
+    assert_file "test/helpers/account_helper_test.rb"
   end
 
   def test_does_not_invoke_helper_if_required
     run_generator ["account", "--skip-helper"]
     assert_no_file "app/helpers/account_helper.rb"
-    assert_no_file "test/unit/helpers/account_helper_test.rb"
+    assert_no_file "test/helpers/account_helper_test.rb"
   end
 
   def test_invokes_default_test_framework
     run_generator
-    assert_file "test/functional/account_controller_test.rb"
+    assert_file "test/controllers/account_controller_test.rb"
   end
 
   def test_does_not_invoke_test_framework_if_required
     run_generator ["account", "--no-test-framework"]
-    assert_no_file "test/functional/account_controller_test.rb"
+    assert_no_file "test/controllers/account_controller_test.rb"
   end
 
   def test_invokes_default_template_engine
diff --git a/railties/test/generators/helper_generator_test.rb b/railties/test/generators/helper_generator_test.rb
index 3d65501..45f3263 100644
--- a/railties/test/generators/helper_generator_test.rb
+++ b/railties/test/generators/helper_generator_test.rb
@@ -14,7 +14,7 @@ class HelperGeneratorTest < GeneratorsTestCase
 
   def test_invokes_default_test_framework
     run_generator
-    assert_file "test/unit/helpers/admin_helper_test.rb", /class AdminHelperTest < ActionView::TestCase/
+    assert_file "test/helpers/admin_helper_test.rb", /class AdminHelperTest < ActionView::TestCase/
   end
 
   def test_logs_if_the_test_framework_cannot_be_found
diff --git a/railties/test/generators/integration_test_generator_test.rb b/railties/test/generators/integration_test_generator_test.rb
index 190b280..83a1aff 100644
--- a/railties/test/generators/integration_test_generator_test.rb
+++ b/railties/test/generators/integration_test_generator_test.rb
@@ -6,7 +6,7 @@ class IntegrationTestGeneratorTest < GeneratorsTestCase
 
   def test_integration_test_skeleton_is_created
     run_generator
-    assert_file "test/integration/integration_test.rb", /class IntegrationTest < ActionController::IntegrationTest/
+    assert_file "test/controllers/integration_test.rb", /class IntegrationTest < ActionController::IntegrationTest/
   end
 
   protected
diff --git a/railties/test/generators/mailer_generator_test.rb b/railties/test/generators/mailer_generator_test.rb
index 9e74859..4f137d1 100644
--- a/railties/test/generators/mailer_generator_test.rb
+++ b/railties/test/generators/mailer_generator_test.rb
@@ -16,7 +16,7 @@ class MailerGeneratorTest < GeneratorsTestCase
 
   def test_invokes_default_test_framework
     run_generator
-    assert_file "test/unit/notifier_test.rb", /class NotifierTest < ActionMailer::TestCase/
+    assert_file "test/models/notifier_test.rb", /class NotifierTest < ActionMailer::TestCase/
     assert_file "test/fixtures/notifier/foo", /app\/views\/notifier\/foo/
     assert_file "test/fixtures/notifier/bar", /app\/views\/notifier\/bar/
   end
diff --git a/railties/test/generators/model_generator_test.rb b/railties/test/generators/model_generator_test.rb
index 501c7d1..cac1327 100644
--- a/railties/test/generators/model_generator_test.rb
+++ b/railties/test/generators/model_generator_test.rb
@@ -152,7 +152,7 @@ class ModelGeneratorTest < GeneratorsTestCase
 
   def test_invokes_default_test_framework
     run_generator
-    assert_file "test/unit/account_test.rb", /class AccountTest < ActiveSupport::TestCase/
+    assert_file "test/models/account_test.rb", /class AccountTest < ActiveSupport::TestCase/
     assert_file "test/fixtures/accounts.yml", /name: MyString/, /age: 1/
   end
 
diff --git a/railties/test/generators/observer_generator_test.rb b/railties/test/generators/observer_generator_test.rb
index a556731..2fd662e 100644
--- a/railties/test/generators/observer_generator_test.rb
+++ b/railties/test/generators/observer_generator_test.rb
@@ -16,7 +16,7 @@ class ObserverGeneratorTest < GeneratorsTestCase
 
   def test_invokes_default_test_framework
     run_generator
-    assert_file "test/unit/account_observer_test.rb", /class AccountObserverTest < ActiveSupport::TestCase/
+    assert_file "test/observers/account_observer_test.rb", /class AccountObserverTest < ActiveSupport::TestCase/
   end
 
   def test_logs_if_the_test_framework_cannot_be_found
diff --git a/railties/test/generators/resource_generator_test.rb b/railties/test/generators/resource_generator_test.rb
index 876ddbf..e4d6908 100644
--- a/railties/test/generators/resource_generator_test.rb
+++ b/railties/test/generators/resource_generator_test.rb
@@ -25,7 +25,7 @@ class ResourceGeneratorTest < GeneratorsTestCase
 
     %w(
       app/models/account.rb
-      test/unit/account_test.rb
+      test/models/account_test.rb
       test/fixtures/accounts.yml
     ).each { |path| assert_file path }
 
@@ -40,10 +40,10 @@ class ResourceGeneratorTest < GeneratorsTestCase
   def test_resource_controller_with_pluralized_class_name
     run_generator
     assert_file "app/controllers/accounts_controller.rb", /class AccountsController < ApplicationController/
-    assert_file "test/functional/accounts_controller_test.rb", /class AccountsControllerTest < ActionController::TestCase/
+    assert_file "test/controllers/accounts_controller_test.rb", /class AccountsControllerTest < ActionController::TestCase/
 
     assert_file "app/helpers/accounts_helper.rb", /module AccountsHelper/
-    assert_file "test/unit/helpers/accounts_helper_test.rb", /class AccountsHelperTest < ActionView::TestCase/
+    assert_file "test/helpers/accounts_helper_test.rb", /class AccountsHelperTest < ActionView::TestCase/
   end
 
   def test_resource_controller_with_actions
@@ -77,14 +77,14 @@ class ResourceGeneratorTest < GeneratorsTestCase
   def test_plural_names_are_singularized
     content = run_generator ["accounts"]
     assert_file "app/models/account.rb", /class Account < ActiveRecord::Base/
-    assert_file "test/unit/account_test.rb", /class AccountTest/
+    assert_file "test/models/account_test.rb", /class AccountTest/
     assert_match /Plural version of the model detected, using singularized version. Override with --force-plural./, content
   end
 
   def test_plural_names_can_be_forced
     content = run_generator ["accounts", "--force-plural"]
     assert_file "app/models/accounts.rb", /class Accounts < ActiveRecord::Base/
-    assert_file "test/unit/accounts_test.rb", /class AccountsTest/
+    assert_file "test/models/accounts_test.rb", /class AccountsTest/
     assert_no_match /Plural version of the model detected/, content
   end
 
diff --git a/railties/test/generators/scaffold_controller_generator_test.rb b/railties/test/generators/scaffold_controller_generator_test.rb
index 834e43e..a6826ae 100644
--- a/railties/test/generators/scaffold_controller_generator_test.rb
+++ b/railties/test/generators/scaffold_controller_generator_test.rb
@@ -48,7 +48,7 @@ class ScaffoldControllerGeneratorTest < GeneratorsTestCase
   def test_helper_are_invoked_with_a_pluralized_name
     run_generator
     assert_file "app/helpers/users_helper.rb", /module UsersHelper/
-    assert_file "test/unit/helpers/users_helper_test.rb", /class UsersHelperTest < ActionView::TestCase/
+    assert_file "test/helpers/users_helper_test.rb", /class UsersHelperTest < ActionView::TestCase/
   end
 
   def test_views_are_generated
@@ -66,7 +66,7 @@ class ScaffoldControllerGeneratorTest < GeneratorsTestCase
   def test_functional_tests
     run_generator
 
-    assert_file "test/functional/users_controller_test.rb" do |content|
+    assert_file "test/controllers/users_controller_test.rb" do |content|
       assert_match /class UsersControllerTest < ActionController::TestCase/, content
       assert_match /test "should get index"/, content
     end
@@ -79,7 +79,7 @@ class ScaffoldControllerGeneratorTest < GeneratorsTestCase
       assert_no_match /def index/, content
     end
 
-    assert_file "test/functional/users_controller_test.rb" do |content|
+    assert_file "test/controllers/users_controller_test.rb" do |content|
       assert_no_match /test "should get index"/, content
     end
 
@@ -89,7 +89,7 @@ class ScaffoldControllerGeneratorTest < GeneratorsTestCase
   def test_skip_helper_if_required
     run_generator ["User", "name:string", "age:integer", "--no-helper"]
     assert_no_file "app/helpers/users_helper.rb"
-    assert_no_file "test/unit/helpers/users_helper_test.rb"
+    assert_no_file "test/helpers/users_helper_test.rb"
   end
 
   def test_skip_layout_if_required
diff --git a/railties/test/generators/scaffold_generator_test.rb b/railties/test/generators/scaffold_generator_test.rb
index 05eadd3..2b0d78a 100644
--- a/railties/test/generators/scaffold_generator_test.rb
+++ b/railties/test/generators/scaffold_generator_test.rb
@@ -19,7 +19,7 @@ class ScaffoldGeneratorTest < GeneratorsTestCase
 
     # Model
     assert_file "app/models/product_line.rb", /class ProductLine < ActiveRecord::Base/
-    assert_file "test/unit/product_line_test.rb", /class ProductLineTest < ActiveSupport::TestCase/
+    assert_file "test/models/product_line_test.rb", /class ProductLineTest < ActiveSupport::TestCase/
     assert_file "test/fixtures/product_lines.yml"
     assert_migration "db/migrate/create_product_lines.rb"
 
@@ -66,7 +66,7 @@ class ScaffoldGeneratorTest < GeneratorsTestCase
       end
     end
 
-    assert_file "test/functional/product_lines_controller_test.rb",
+    assert_file "test/controllers/product_lines_controller_test.rb",
                 /class ProductLinesControllerTest < ActionController::TestCase/
 
     # Views
@@ -80,7 +80,7 @@ class ScaffoldGeneratorTest < GeneratorsTestCase
 
     # Helpers
     assert_file "app/helpers/product_lines_helper.rb"
-    assert_file "test/unit/helpers/product_lines_helper_test.rb"
+    assert_file "test/helpers/product_lines_helper_test.rb"
 
     # Stylesheets
     assert_file "public/stylesheets/scaffold.css"
@@ -92,7 +92,7 @@ class ScaffoldGeneratorTest < GeneratorsTestCase
 
     # Model
     assert_no_file "app/models/product_line.rb"
-    assert_no_file "test/unit/product_line_test.rb"
+    assert_no_file "test/models/product_line_test.rb"
     assert_no_file "test/fixtures/product_lines.yml"
     assert_no_migration "db/migrate/create_product_lines.rb"
 
@@ -103,7 +103,7 @@ class ScaffoldGeneratorTest < GeneratorsTestCase
 
     # Controller
     assert_no_file "app/controllers/product_lines_controller.rb"
-    assert_no_file "test/functional/product_lines_controller_test.rb"
+    assert_no_file "test/controllers/product_lines_controller_test.rb"
 
     # Views
     assert_no_file "app/views/product_lines"
@@ -111,7 +111,7 @@ class ScaffoldGeneratorTest < GeneratorsTestCase
 
     # Helpers
     assert_no_file "app/helpers/product_lines_helper.rb"
-    assert_no_file "test/unit/helpers/product_lines_helper_test.rb"
+    assert_no_file "test/helpers/product_lines_helper_test.rb"
 
     # Stylesheets (should not be removed)
     assert_file "public/stylesheets/scaffold.css"
-- 
1.6.4

